### ISSUE TITLE
Fontstand download URL now does User Agent checking

### ISSUE THREAD

**Issue by dwaite** [caskroom/homebrew-cask#11349]
The download URL within the Fontstand app now does User-Agent Header checking apparently, it returns a HTML page saying the app is only supported on Mac OS X instead of the binary.

I worked around this by changing the url to the following line
url 'https://fontstand.com/application/download', :user_agent => "Mozilla/5.0 (Macintosh; Intel Mac OS X 10) http://caskroom.io"

**Comment by vitorgalvao** [comment]
Does it work if you use `:user_agent => :fake`? Care to submit a PR?

**Comment by dwaite** [comment]
:fake seems to only say "Chrome", it doesn't give a full-on browser agent string.

Is there specific rationale around the contents of the default user agent or :fake agent? 

I'd be fine doing a PR, but I'd prefer to fix one of the built-ins (or add a third) rather than hard-code a cask

**Comment by vitorgalvao** [comment]
`:fake` works most of the time, so if it works here, it should be the one used.

As for implementing new ones, pinging @phinze @ndr-qef.

**Comment by dwaite** [comment]
Investigation:

User Agent strings from browsers on my machine:

* Firefox 37.0.1:

     ```
      User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10.10; rv:37.0) Gecko/20100101 Firefox/37.0
     ```
* Safari 8.0.7:

   ```
      User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_10_4) AppleWebKit/600.7.5 (KHTML, like Gecko) Version/8.0.7 Safari/600.7.5
   ```
* Chrome 43.0:

   ```
      User-Agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_10_4) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/43.0.2357.65 Safari/537.36
   ```

vs the default agent for cask asset downloads:
```
User-Agent: Homebrew-cask v0.51+ (Ruby 2.0.0-481)
```

or the :fake agent:
```
User-Agent: Chrome/32.0.1000.00
```
The browser strings are crafted specifically to match the unfortunate server page rendering and local javascript regular expressions out there:
* Starts with `Mozilla/5.0`
* Contains `(Macintosh; Intel Mac OS X 10 ...)`
* Contains the string "Gecko"
* Includes platform and browser version strings so that one could gather statistics about the traffic hitting the site

I might propose instead a default agent of
```
Mozilla/5.0 (Macintosh; Intel Mac OS X 10.10) Caskroom/v0.51+, homebrew-casks/db9991 unlike Gecko
```

This includes the system platform and version (which may actually be useful for statistics and filtering), reports Both that it is caskroom and that it is operating off of the homebred-casks tap, including versions of both (the homebred-casks version is the first six hex characters of the git commit), and finally starts with Mozilla and trails with Gecko, in case any regular expression would fail without them.

in this case, :fake would be - any of the browser user agents at the top. I prefer Firefox's since its shortest.

**Comment by vitorgalvao** [comment]
I have absolutely no opposition to that. Sounds OK to me. Lets wait for @phinze and @ndr-qef to reply to see if we’re all on the same page.

**Comment by ndr-qef** [comment]
> Is there specific rationale around the contents of the default user agent or :fake agent?

No, I guess it was the simplest thing which happened to work at the time.

I would prefer not to include misleading information in our default user agent; after all, user agent woes are rare. However, improvements to `:fake` would be welcome.

**Comment by phinze** [comment]
:+1: this all sounds reasonable to me

**Comment by ndr-qef** [comment]
Closed by #11995, which was fashioned after @dwaite's suggestion. Thanks.
